### PR TITLE
fix : events from call info / UDC cairo 0 migration

### PR DESCRIPTION
**Pull request by ocdbytes**
- Events emitting fix.
- UDC migration to cairo 0. (to support starknet.js `deploy` function)

Pending : 

- need to add unit test cases.

### CODE REVIEW

File: crates/primitives/receipt/src/from_blockifier.rs [hunk 1]
```diff
@@ -134,8 +126,128 @@ pub fn from_blockifier_execution_info(res: &TransactionExecutionInfo, tx: &Trans
...
+    fn test_event_ordering() {
+        let mut events = Vec::new();
+        get_events_from_call_info(Some(&call_info()), 0, &mut events);
+
+        let expected_events_ordering = vec![event(0), event(1), event(2), event(3), event(4)];
+
+        assert_eq!(expected_events_ordering, events);
+    }
+
+    fn call_info() -> CallInfo {
```
Comment (apoorvsadana) [inline crates/primitives/receipt/src/from_blockifier.rs]: does this work (AI generated)


```rust
fn create_call_info(event_number: u32, inner_calls: Vec<CallInfo>) -> CallInfo {
    CallInfo {
        call: Default::default(),
        execution: execution(vec![ordered_event(event_number)]),
        resources: Default::default(),
        inner_calls,
        storage_read_values: vec![],
        accessed_storage_keys: Default::default(),
    }
}

let nested_calls = vec![
    create_call_info(1, vec![
        create_call_info(2, vec![
            create_call_info(3, vec![
                create_call_info(4, vec![])
            ])
        ])
    ])
];
```

Comment (Mohiiit) [reply crates/primitives/receipt/src/from_blockifier.rs]: resolved

**Review by apoorvsadana — APPROVED** [review]

File: crates/primitives/receipt/Cargo.toml [hunk 2]
```diff
@@ -19,11 +19,13 @@ mp-convert = { workspace = true }
...
 blockifier = { workspace = true }
 cairo-vm = { workspace = true }
+rstest.workspace = true
 starknet-core = { workspace = true }
 starknet-providers = { workspace = true }
 starknet-types-core = { workspace = true }
-
+starknet_api = { workspace = true }
 # Other
+log = "0.4.21"
```
Comment (Trantorian1) [inline crates/primitives/receipt/Cargo.toml]: Please use workspace log

**Review by Trantorian1 — CHANGES_REQUESTED** [review]
All seems good, just please update your use of the `log` crate to refer to workspace dependencies.

**Review by EvolveArt — APPROVED** [review]

File: crates/primitives/receipt/Cargo.toml (hunk 2 above)
Comment (Mohiiit) [reply crates/primitives/receipt/Cargo.toml]: resolved

**Review by Trantorian1 — APPROVED** [review]

**Pull request merged**